packages/elements-core/src/hooks/useParsedData.ts

```diff
@@ -3,11 +3,20 @@ import { NodeType } from '@stoplight/types';
 import { parse as parseYaml } from '@stoplight/yaml';
 import * as React from 'react';
 
-import { JSONSchema, ParsedNode } from '../types';
+import { JSONSchema, ParsedNode, UnparsedNode } from '../types';
 import { isHttpOperation, isHttpService, isJSONSchema, isSMDASTRoot } from '../utils/guards';
 
-export function useParsedData(nodeType: string, data: unknown): ParsedNode | undefined {
-  return React.useMemo(() => parserMap[nodeType]?.(data), [nodeType, data]);
+export function useParsedData(
+  unparsedNode: UnparsedNode | undefined,
+  fallbackNode?: ParsedNode,
```

Comment:
This "fallback" that isn't really a fallback (as it takes priority) feels off here. More thoughts above.

Reply:
This really comes from the fact that hooks suck.

In reagular code I would just make a readable condition and this would never ever be a problem.

Hooks have no proper support for conditionals so you end up with code smell like that.